kubejs/server_scripts/tconstruct.js

```diff
@@ -147,4 +151,52 @@ ServerEvents.recipes((event) => {
             .mixing(Fluid.of(output, FluidAmounts.NUGGET), ingMix)
             .heated();
     }
+    
+    function addMMT(material_name, ingot_melt_ticks, ingot_cool_ticks, temperature, crushing_nuggets) {
+        let nugget_melt_ticks = ingot_melt_ticks / 9;
+        let block_melt_ticks = ingot_melt_ticks * 9;
+        let nugget_cool_ticks = ingot_cool_ticks / 9;
+        let block_cool_ticks = ingot_cool_ticks * 9;
+        let liquid = "grand_expanse:molten_" + material_name;
+        
+        let raw = "mmt:raw_" + material_name;
+        melt(raw, liquid, FluidAmounts.INGOT, ingot_melt_ticks, temperature);
+        
+        let dust = "mmt:" + material_name + "_dust";
+        melt(dust, liquid, FluidAmounts.INGOT, ingot_melt_ticks, temperature);
+
+        let rawBlock = "mmt:raw_" + material_name + "_block";
+        melt(rawBlock, liquid, FluidAmounts.METAL_BLOCK, block_melt_ticks, temperature);
+        
+        let nugget = "mmt:" + material_name + "_nugget";
+        melt(nugget, liquid, FluidAmounts.INGOT, nugget_melt_ticks, temperature);
+        cast_type(liquid, FluidAmounts.INGOT, nugget, "nugget", nugget_cool_ticks);
+        
+        let ingot = "mmt:" + material_name + "_ingot";
+        melt(ingot, liquid, FluidAmounts.INGOT, ingot_melt_ticks, temperature);
+        cast_type(liquid, FluidAmounts.INGOT, ingot, "ingot", ingot_cool_ticks);      
+        
+        let plate = "mmt:" + material_name + "_plate";
+        melt(plate, liquid, FluidAmounts.INGOT, ingot_melt_ticks, temperature);
+        cast_type(liquid, FluidAmounts.INGOT, plate, "plate", ingot_cool_ticks);
+
+        let block = "mmt:" + material_name + "_block";
+        melt(block, liquid, FluidAmounts.METAL_BLOCK, block_melt_ticks, temperature);
+        pour(liquid, FluidAmounts.METAL_BLOCK, block, block_cool_ticks, true);
+        
+        let gear = "mmt:" + material_name + "_gear";
+        melt(gear, liquid, FluidAmounts.INGOT * 4, ingot_melt_ticks * 4, temperature);
+        cast_type(liquid, FluidAmounts.INGOT * 4, gear, "gear", ingot_cool_ticks * 4);
+                
+        let rod = "mmt:" + material_name + "_rod";
+        melt(rod, liquid, FluidAmounts.INGOT / 2, ingot_melt_ticks / 2, temperature);
+        cast_type(liquid, FluidAmounts.INGOT / 2, rod, "rod", ingot_cool_ticks / 2);
+
+        let wire = "mmt:" + material_name + "_wire";
+        melt(wire, liquid, FluidAmounts.INGOT / 2, ingot_melt_ticks / 2, temperature);
+        cast_type(liquid, FluidAmounts.INGOT / 2, wire, "wire", ingot_cool_ticks / 2);
+        
+        event.recipes.create.crushing(dust, ingot);
+        event.recipes.create.crushing([dust, Item.of(nugget, crushing_nuggets)], raw);
```

Comment:
don't really like having this here but we can keep it until we need to add more general create functions.